src/Pos/Block/Logic.hs

```diff
@@ -829,12 +829,11 @@ createMainBlockFinish slotId pSk prevHeader = do
         let convertTx (txId, (tx, _, _)) = WithHash tx txId
         sortedTxs <- maybe onBrokenTopo pure $ topsortTxs convertTx localTxs
         sk <- npSecretKey . ncNodeParams <$> getNodeContext
-        -- for now let's be cautious and not generate blocks that are larger than
-        -- maxBlockSize/4
-        sizeLimit <- fromIntegral . toBytes . (`div` 4) <$> UDB.getMaxBlockSize
+        sizeLimit <- fromIntegral . toBytes <$> UDB.getMaxBlockSize
         block <- createMainBlockPure
-                     sizeLimit prevHeader sortedTxs pSk
-                     slotId localPSKs sscData usPayload sk
+            sizeLimit prevHeader sortedTxs pSk
+            slotId localPSKs sscData usPayload sk
+        lift $ logInfo $ "Created main block of size: " <> show (length $ Bi.encode block)
```

Comment:
Better use `biSize`, because most likely with `store` `biSize` will be more effective than `length . encode`.